backend/gateway/stats/runtime.go

```diff
@@ -0,0 +1,77 @@
+package stats
+
+import (
+	"context"
+	"runtime"
+	"time"
+
+	"github.com/uber-go/tally"
+)
+
+type RuntimeStatCollector struct {
+	scope tally.Scope
+}
+
+func NewRuntimeStats(scope tally.Scope) *RuntimeStatCollector {
+	return &RuntimeStatCollector{
+		scope: scope.SubScope("runtime"),
+	}
+}
+
+func (r *RuntimeStatCollector) Collect(ctx context.Context) {
+	ticker := time.NewTicker(time.Second * 10)
+	for {
+		r.collectCPUStats()
+		r.collectMemStats()
+		select {
+		case <-ticker.C:
+			continue
+		case <-ctx.Done():
+			ticker.Stop()
+			return
+		}
+	}
+}
+
+func (r *RuntimeStatCollector) collectCPUStats() {
+	r.scope.Gauge("cpu.goroutines").Update(float64(runtime.NumGoroutine()))
+	r.scope.Gauge("cpu.cgo_calls").Update(float64(runtime.NumCgoCall()))
+}
+
+func (r *RuntimeStatCollector) collectMemStats() {
+	var memStats runtime.MemStats
+	runtime.ReadMemStats(&memStats)
+
+	// General Stats
+	r.scope.Gauge("memory.alloc").Update(float64(memStats.Alloc))
```

Comment:
you should predefine all of the gauges in a struct e.g.

```go
type runtimeMetrics struct {
   memoryAlloc tally.Gauge
}
```

and then set it up in New()